pkg/retention/intervalCalender.go

```diff
@@ -0,0 +1,47 @@
+package retention
+
+import (
+	"fmt"
+)
+
+// intervalCalendar stores the time intervals of an interval-based retention strategy and validates the configuration
+type intervalCalendar struct {
+	name          StrategyId
+	timeIntervals []timeInterval
+}
+
+// invalidCalendarConfigError contains the message and intervalName of an invalid retention strategy
+//
+//nolint:unused
```

Comment:
This struct is used.

Reply:
Well yes, but only in `validateConfig()` which currently is only used in tests. Therefore, the linter will mark it as unused.